ipatests/test_integration/test_caless.py

```diff
@@ -309,23 +306,45 @@ def get_replica_filename(self, replica):
                             'replica-info.gpg')
 
     @classmethod
-    def export_pkcs12(cls, nickname, filename='server.p12', password=None):
-        """Export a cert as PKCS#12 to the given file"""
+    def create_pkcs12(cls, nickname, filename='server.p12', password=None):
+        """Create a cert chain and generate pkcs12 cert"""
         if password is None:
             password = cls.cert_password
-        ipautil.run(['pk12util',
-                     '-o', filename,
-                     '-n', nickname,
-                     '-d', 'nssdb',
-                     '-K', cls.cert_password,
-                     '-W', password], cwd=cls.cert_dir)
+
+        fname_chain = []
+
+        key_fname = os.path.join(cls.cert_dir, nickname + '.key')
+        cert_fname = os.path.join(cls.cert_dir, nickname + '.crt')
+
+        nick_chain = nickname.split('/')
+
+        # to construct whole chain e.g "ca1 - ca1/sub - ca1/sub/server"
+        for index, _value in enumerate(nick_chain):
```

Comment:
Usually, you'd probably just do `for idx in range(0, len(nick_chain))` but I don't mind this staying here.